cloudflare/resource_cloudflare_access_policy.go

```diff
@@ -66,8 +66,70 @@ func resourceCloudflareAccessPolicy() *schema.Resource {
 				Required: true,
 				Elem:     AccessGroupOptionSchemaElement,
 			},
+			"purpose_justification_required": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"purpose_justification_prompt": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"purpose_justification_required"},
+			},
+			"approval_group": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     AccessPolicyApprovalGroupElement,
+			},
+		},
+	}
+}
+
+var AccessPolicyApprovalGroupElement = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"email_list_uuid": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"email_addresses": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+		"approvals_needed": {
+			Type:         schema.TypeInt,
+			Required:     true,
+			ValidateFunc: validation.IntAtLeast(0),
 		},
+	},
+}
+
+func APIAccessPolicyApprovalGroupToSchema(approvalGroup cloudflare.AccessApprovalGroup) map[string]interface{} {
```

Comment:
I don't think we want this method exported publicly; best to lowercase it.

```suggestion
func apiAccessPolicyApprovalGroupToSchema(approvalGroup cloudflare.AccessApprovalGroup) map[string]interface{} {
```